File: leetcode/169.cpp
class Solution {
public:
    int majorityElement(vector<int> &num) {
        map<int, int> mm;
        int n = num.size();
        for (int i = 0; i < num.size(); i++) {
            if (++mm[num[i]] > n / 2)
                return num[i];
        }
    }
};

File: leetcode/169.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def majorityElement(self, nums):
        if len(nums) == 0:
            return []
        nownum = nums[0]
        nowcnt = 1
        for i in range(1, len(nums)):
            if nowcnt == 0:
                nownum = nums[i]
                nowcnt = 1
            elif nownum != nums[i]:
                nowcnt -= 1
            else:
                nowcnt += 1
        return nownum
